cli-plugins/manager/metadata.go

```diff
@@ -0,0 +1,19 @@
+package manager
+
+const (
+	// NamePrefix is the prefix required on all plugin binary names
+	NamePrefix = "docker-"
+
+	// MetadataSubcommandName is the name of the plugin subcommand
+	// which must be supported by every plugin and returns the
+	// plugin metadata.
+	MetadataSubcommandName = "docker-cli-plugin-metadata"
+)
+
+// Metadata provided by the plugin
+type Metadata struct {
```

Comment:
As a follow-up would perhaps be nice to have a json-schema for this to define the spec. This could help plugin-authors to write a plugin in anything else than Go 🤔

Reply:
There is an _informal_ schema in `docs/extend/cli_plugins.md` (and right now there aren't many fields nor are they complex) but I do agree with you.